src/os/bluestore/BlueStore.cc

```diff
@@ -15297,22 +15292,29 @@ int BlueStore::_deferred_replay()
     bool has_some = _eliminate_outdated_deferred(deferred_txn, bluefs_extents);
     if (has_some) {
       if (debug_deferred_replay_track) debug_deferred_replay_track(*deferred_txn);
-      TransContext *txc = _txc_create(ch.get(), osr,  nullptr);
-      txc->deferred_txn = deferred_txn;
-      txc->set_state(TransContext::STATE_KV_DONE);
-      _txc_state_proc(txc);
+      for (auto& op: deferred_txn->ops) {
+        for (auto& e : op.extents) {
+          bufferlist t;
+          op.data.splice(0, e.length, &t);
+          bdev->aio_write(e.offset, t, &ioctx, false);
+        }
+      }
     } else {
       delete deferred_txn;
     }
   }
  out:
-  dout(20) << __func__ << " draining osr" << dendl;
-  _osr_register_zombie(osr);
-  _osr_drain_all();
-  if (debug_deferred_replay_end) debug_deferred_replay_end();
-  if (fake_ch) {
-    new_coll_map.clear();
+  bdev->aio_submit(&ioctx);
```

Comment:
Are we sure that it's safe enough to try to accumulate all the deferred writes in a single IO/DB txc batch? Couldn't it go beyond some bounds?

Reply:
It is limited by amount of deferred writes pending. It is limited by amount of deferred writes BlueStore queued before crash.
Plus ones that are stale. 

I will take a look.

Reply:
Added batching of bdev ops.